src/lolex-src.js

```diff
@@ -21,6 +21,7 @@ function withGlobal(_global) {
         hrtimePresent && typeof _global.process.hrtime.bigint === "function";
     var nextTickPresent =
         _global.process && typeof _global.process.nextTick === "function";
+    var utilPromisify = _global.process && require("util").promisify;
```

Comment:
`_global.process` as a "I am in node" check?

Reply:
Just be sure we are targetting the right `global` here ... Should not this strictly be `globalObject` (as that's where the promisify function will be required from in the runtime)? It's not certain that people will pass a custom global with all of the props of the runtime.

Reply:
> `_global.process` as a "I am in node" check?

Yeah this seems to be one of the popular ways
https://stackoverflow.com/questions/4224606/how-to-check-whether-a-script-is-running-under-node-js

> Just be sure we are targetting the right `global` here ... Should not this strictly be `globalObject` (as that's where the promisify function will be required from in the runtime)? It's not certain that people will pass a custom global with all of the props of the runtime.

TBH I don't think I have the necessary knowledge of what the use cases for non-`defaultImplementation`s / the `withGlobal` function are to judge whether there will be cases where the provided `_global` looks like a Node environment, but our actual global scope does not have `require`. However FWIW I think it would be unintuitive to have vastly different conditions for applying e.g. the `nextTick` and `util.promisify` special handling, so that's why I'd prefer this.

Reply:
`withGlobal` was added for jest support so we can pass in the jsdom global.

Reply:
So then it'll apply to the `jest-environment-jsdom` case anyway because it has `process`.
Imagining that I pass in e.g. a DOM global that does *not* have Node things like `process`, I would probably want it to not apply the promisify change (even though it could, using the outside `require`), just like it doesn't apply the `nextTick` change, I think?

Reply:
I think inspecting `_global` is correct - if the global used to instantiate `lolex` pretends to be from node (by having a `process` property) we should assume we can `require('util');` safely and go with that.